lib/videoroom/metrics_persistor.ex

```diff
@@ -105,18 +114,39 @@ defmodule VideoRoom.MetricsPersistor do
           {key, value}
       end)
 
-    report
-    |> Map.take(@metrics_to_derive)
-    |> Enum.map(fn {key, value} when is_number(value) ->
-      case get_in(state, [:prev_report | path ++ [key]]) do
-        old_value when is_number(old_value) ->
-          derivative = (value - old_value) * 1000 / (time - state.prev_report_ts)
-          {"#{key}-per-second", derivative}
+    report =
+      report
+      |> Map.take(@metrics_to_derive)
+      |> Enum.map(fn {key, value} when is_number(value) ->
+        case get_in(state, [:prev_report | path ++ [key]]) do
+          old_value when is_number(old_value) ->
+            derivative = (value - old_value) * 1000 / (time - state.prev_report_ts)
+            {"#{key}-per-second", derivative}
+
+          _otherwise ->
+            nil
+        end
+      end)
+      |> Enum.reject(&is_nil/1)
+      |> Enum.into(report)
+
+    for {m1, m2, res} <- @compound_metrics_to_derive do
+      case Map.take(report, [m1, m2]) do
+        %{^m1 => m1_v, ^m2 => m2_v} ->
+          old_m1_v = get_in(state, [:prev_report | path ++ [m1]])
+          old_m2_v = get_in(state, [:prev_report | path ++ [m2]])
+
+          if is_nil(old_m1_v) or is_nil(old_m2_v) do
+            nil
+          else
+            metric = (m1_v - old_m1_v) / (m2_v - old_m2_v)
+            {"#{res}", metric}
+          end
```

Comment:
It should be fine to generate the first report as well by using 0 as a default. More importantly, it will remove the if :wink: 

```suggestion
          old_m1_v = get_in(state.prev_report, path ++ [m1]]) || 0
          old_m2_v = get_in(state.prev_report, path ++ [m2]]) || 0

          metric = (m1_v - old_m1_v) / (m2_v - old_m2_v)
          {"#{res}", metric}
```
Or with single `get_in`:
```suggestion
          old_values = get_in(state.prev_report, path) || %{}
          old_m1_v = Map.get(old_values, m1, 0)
          old_m2_v = Map.get(old_values, m2, 0)

          metric = (m1_v - old_m1_v) / (m2_v - old_m2_v)
          {"#{res}", metric}
```